src/keycloak/chart/templates/pvc.yaml

```
@@ -22,8 +82,15 @@ metadata:
     {{- include "keycloak.labels" . | nindent 4 }}
 spec:
   accessModes:
-    - ReadWriteOnce
+    - {{ .Values.persistence.accessMode | quote }}
   resources:
     requests:
-      storage: 512Mi
+      storage: {{ .Values.persistence.themes.size | quote }}
+  {{ if .Values.persistence.storageClassName }}
+  {{- if (eq "-" .Values.persistence.storageClassName) }}
+  storageClassName: ""
+  {{- else }}
+  storageClassName: "{{ .Values.persistence.storageClassName }}"
+  {{- end }}
```

Think I'm a bit confused on the template here - what is the intent of the `-` check? `storageClassName: ""` would be equivalent to no storage class I believe? Is no storage class an actual need here, or can we just simplify the template to:
```suggestion
  {{ if .Values.persistence.storageClassName }}
  storageClassName: "{{ .Values.persistence.storageClassName }}"
  {{- end }}
```

---

It's probably overcomplicated.. but there is technically a difference between an emptystring storageClassName and a nil/non-existent one. In theory, you might want to explicitly set `storageClassName: ""` to prevent automatic provisioning, which is what the special `"-"` case is meant to handle. Not providing `storageClassName` at all results in using the default SC.

I'm not sure I've ever found myself in that particular situation, but it's a fairly common [pattern I've seen elsewhere](https://github.com/bitnami/charts/blob/main/bitnami/common/templates/_storage.tpl#L15).

---

Yep - was tracking the difference just wasn't sure if we needed to add that support here (we already assume storage class for other charts that I don't think support this). All good with leaving though if you think it's a value add.

---

It's probably overkill in this case.. #deletedeletedelete 😉